### PR TITLE
Initial pass: Pug conversion

### PR DESCRIPTION
**Pull request by kkoskelin**
Introduction of pug/sass templating at a high level within `shared/src/business/utilities/generateHTMLTemplateForPDF.js`
Plenty of other conversion will be needed, but this is a starting point.  Introducing these functions also means ensuring calls are (de)structured and include `applicationContext`, as well as using `await`.

### CODE REVIEW

File: shared/src/business/utilities/generateHTMLTemplateForPDF.js [hunk 1]
```diff
@@ -3,163 +3,50 @@ const { capitalize } = require('lodash');
...
-            ${options.h2 ? '<h2>' + options.h2 + '</h2>' : ''}
-            <div class="clear"></div>
-          </div>
+const generateHTMLTemplateForPDF = async ({
+  applicationContext,
+  content,
+  options = {},
+}) => {
+  const sassContent = require('./htmlGenerator/index.scss_');
+  const template = require('./htmlGenerator/index.pug_');
```
Comment (rachaelparris) [inline shared/src/business/utilities/generateHTMLTemplateForPDF.js:18]: Why the underscores at the end of the file names here?

**Review by rachaelparris — APPROVED** [review]
Looks awesome.

File: shared/src/business/utilities/generateHTMLTemplateForPDF.js (hunk 1 above)
Comment (kkoskelin) [reply shared/src/business/utilities/generateHTMLTemplateForPDF.js:18]: It is actually referring to "module-ified" static resources created when you call "build:assets". The underscore naming convention is up for debate, happy to discuss.

Comment (rachaelparris) [reply shared/src/business/utilities/generateHTMLTemplateForPDF.js:18]: Ah, I see. Makes sense. 👍

**Pull request merged**